modules/bluebillywigBidAdapter.js

```diff
@@ -0,0 +1,490 @@
+import * as utils from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import adapterManager from '../src/adapterManager.js';
+import { VIDEO } from '../src/mediaTypes.js';
+import { config } from '../src/config.js';
+import { Renderer } from '../src/Renderer.js';
+
+const DEV_MODE = window.location.search.match(/bbpbs_debug=true/);
+
+// Blue Billywig Constants
+export const BB_CONSTANTS = {
+  BIDDER_CODE: 'bluebillywig',
+  AUCTION_URL: '$$URL_STARTpbs.bluebillywig.com/openrtb2/auction?pub=$$PUBLICATION',
+  SYNC_URL: '$$URL_STARTpbs.bluebillywig.com/static/cookie-sync.html?pub=$$PUBLICATION',
+  RENDERER_URL: 'https://$$PUBLICATION.bbvms.com/r/$$RENDERER.js',
+  DEFAULT_TIMEOUT: 5000,
+  DEFAULT_TTL: 300,
+  DEFAULT_WIDTH: 768,
+  DEFAULT_HEIGHT: 432,
+  DEFAULT_NET_REVENUE: true
+};
+
+// Aliasing
+const getConfig = config.getConfig;
+
+// Helper Functions
+export const BB_HELPERS = {
+  addSiteAppDevice: (request, pageUrl) => {
+    if (!request) return;
+
+    if (typeof getConfig('app') === 'object') request.app = getConfig('app');
+    else if (pageUrl) request.site = { page: pageUrl };
+
+    if (typeof getConfig('device') === 'object') request.device = getConfig('device');
+    if (!request.device) request.device = {};
+    if (!request.device.w) request.device.w = window.innerWidth;
+    if (!request.device.h) request.device.h = window.innerHeight;
+  },
+  addSchain: (request, validBidRequests) => {
+    if (!request) return;
+
+    const schain = utils.deepAccess(validBidRequests, '0.schain');
+    if (schain) request.source.ext = { schain: schain };
+  },
+  addAliases: (request, aliases) => {
+    if (!request) return;
+
+    if (!utils.isEmpty(aliases)) request.ext.prebid.aliases = aliases;
+  },
+  addCurrency: (request) => {
+    if (!request) return;
+
+    const adServerCur = getConfig('currency.adServerCurrency');
+    if (adServerCur && typeof adServerCur === 'string') request.cur = [adServerCur];
+    else if (Array.isArray(adServerCur) && adServerCur.length) request.cur = [adServerCur[0]];
+  },
+  addUserIds: (request, validBidRequests) => {
+    if (!request) return;
+
+    // NB straight rip from prebidServerAdapter, keep track for updates
+    const bidUserId = utils.deepAccess(validBidRequests, '0.userId');
+
+    if (bidUserId && typeof bidUserId === 'object' && (bidUserId.tdid || bidUserId.pubcid || bidUserId.parrableid || bidUserId.lipb || bidUserId.id5id || bidUserId.criteoId || bidUserId.britepoolid || bidUserId.idl_env)) {
```

Comment:
Instead constructing eids one by one, you can just use this function:
https://github.com/prebid/Prebid.js/blob/c0c8b58e4c8622c501fab83a8dd6ef3fe4fa3eec/modules/userId/eids.js#L120

Reply:
Didn't know, cheers! Will fix.

Reply:
Rewrote to use createEidsArray